R/dice.R

```diff
@@ -188,30 +207,21 @@ dice <- function(x, fm = NULL, SPC = TRUE, pctMissing = FALSE, fill = FALSE, str
   # expand slice IDs (horizon IDs)
   # these are used to join with horizons
   sliceIDs <- rep(
-    h[[hzidn]], 
-    times = h[[htb[2]]] - h[[htb[1]]]
+    h.sub[[hzidn]], 
+    times = h.sub[[htb[2]]] - h.sub[[htb[1]]]
   )
   
-  
   # assemble slice LUT for JOIN
   s <- data.table(
     sliceID = sliceIDs, 
-    .sliceTop = tops, 
-    .sliceBottom = bottoms
+    .sliceTop = tops[seq_along(sliceIDs)], 
```

Comment:
Short description of why the new indexing?

Reply:
`mapply()` and `.seq.safe()` were returning non-conforming vectors. Will review this to make sure it is still needed

Reply:
This also was not needed, as filtering 0 thickness horizons is sufficient